### PR TITLE
Phase2-TB63Z Rearrange the header file and make first example for dd4hep (replacing #34752)

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Rearrange the header file in SimG4CMS/HGCalTestBeam and make first example for dd4hep (replacing #34752)

#### PR validation:

Use the cfg's in SimG4CMS/HGCalTestBeam/test

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc [hunk 1]
```diff
@@ -4,12 +4,13 @@
 // Description: Main analysis class for HGCal Validation of G4 Hits
 ///////////////////////////////////////////////////////////////////////////////
 
-#include "HGCPassive.h"
+#include "SimG4CMS/HGCalTestBeam/interface/HGCPassive.h"
```
Comment (civanch) [inline SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc:7]: @bsunanda , I am sorry, but recent requirement from @smuzaffar was to have include files for plugin inside plugin directory. This include files should not be used by any other class except producer itself HGCPassive.cc, normally these two files should be merged together into HGCPassive.cc, two separate files only if the producer is too big.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@srimanob Please approve this

**Comment by srimanob** [comment]
+Upgrade

Technical PR for the header file rearranging, and include new example for dd4hep. PR test runs fine without change as expected.

**Comment by perrotta** [comment]
+1

**Pull request merged**